app/scenes/channel_drawer/channel_drawer.js

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React from 'react';
+import {View} from 'react-native';
+
+import ChannelList from 'app/components/channel_drawer/channel_list';
+
+export default class ChannelDrawer extends React.PureComponent {
+    static propTypes = {
+        actions: React.PropTypes.shape({
+            selectChannel: React.PropTypes.func.isRequired,
+            viewChannel: React.PropTypes.func.isRequired,
+            closeDMChannel: React.PropTypes.func.isRequired,
+            closeDrawers: React.PropTypes.func.isRequired
+        }).isRequired,
+        currentTeam: React.PropTypes.object,
+        currentChannel: React.PropTypes.object,
+        channels: React.PropTypes.object,
+        channelMembers: React.PropTypes.object,
+        theme: React.PropTypes.object.isRequired
+    };
+
+    selectChannel = (id) => {
+        this.props.actions.selectChannel(id);
+        this.props.actions.closeDrawers();
+    }
+
+    render() {
+        const {
+            currentChannel,
+            currentTeam,
+            channels,
+            channelMembers,
+            theme
+        } = this.props;
+
+        return (
+            <View
+                style={{
```

Comment:
I prefer to use flex and maybe calculate some of it cause this may hurt on different screen sizes